### PR TITLE
Fix/caching

### PR DESCRIPTION
**Pull request by Alexanderlacuna**
Pr fixes issue https://github.com/genenetwork/genenetwork2/issues/677.where empty text files results to json decoding error.This
should never happen unless the cache isn't working at all

(TODO)regression tests

### CODE REVIEW

**Comment by zsloan** [comment]
Ah, this is a good solution. In this case I don't think the JSON file was empty (it said it had an error at position 600000-something), but this should still resolve the issue of a flawed cache file by removing it and regenerating a new one.

File: wqflask/wqflask/correlation/pre_computes.py [hunk 1]
```diff
@@ -6,29 +6,40 @@
...
 
+from json.decoder import JSONDecodeError
+
 
 def fetch_all_cached_metadata(dataset_name):
     """in a gvein dataset fetch all the traits metadata"""
     file_name = generate_filename(dataset_name, suffix="metadata")
 
-    file_path = os.path.join(TMPDIR, file_name)
+    file_path = Path(TMPDIR, file_name)
```
Comment (BonfaceKilz) [inline wqflask/wqflask/correlation/pre_computes.py:16]: **Question**: What's the preference of this over "os.path.join"? Here I'm just being curious.

Comment (BonfaceKilz) [reply wqflask/wqflask/correlation/pre_computes.py:16]: Actually, I later see that you use the "unlink" and "touch" so :+1:

**Review by BonfaceKilz — APPROVED** [review]
Nothing looks off to me. Thanks for your contributions and acting promptly on https://github.com/genenetwork/genenetwork2/issues/677

**Comment by Alexanderlacuna** [comment]
going ahead to merge this

**Pull request merged**